Add method that tells the app/backend that we are using a dark mode theme

gcasa opened the pull request:
The purpose of this PR is to add a method to make the system aware that a theme using "dark mode" is being used.   This will assist in settings for editors and, particularly, with printing.

gcasa committed:
Add method that tells the app/backend that we are using a dark mode theme

gcasa committed:
Add method that tells the app/backend that we are using a dark mode theme

gcasa committed:
Add method that tells the app/backend that we are using a dark mode theme

gcasa committed:
Add method that tells the app/backend that we are using a dark mode theme

gcasa committed:
Use GSThemeDarkMode setting in the theme as the return value instead of it being hard coded

gcasa committed:
Use GSThemeDarkMode setting in the theme as the return value instead of it being hard coded

gcasa committed:
Merge branch 'master' of github.com:gnustep/libs-gui into GSTheme_darkmode_branch

gcasa committed:
Merge branch 'master' into GSTheme_darkmode_branch

rmottola commented:
A discussed via voice, I think we need two properties:
- Theme "is dark". When switched system issues dark mode notification and identifies as darkmode.
- Theme supports both light and dark modes

The latter porperty is for future work, non need to implement it now.

isDarkMode in this context is a little bit ambiguous.

rmottola committed:
Split information of what a Theme supports from what is actually being
run. In the future, like Aqua, we might support Themes having both Light
and Dark Appearance.
Change wording from Mode to Appearance, since Apple has NSAppearance.

rmottola committed:
Split information of what a Theme supports from what is actually being
run. In the future, like Aqua, we might support Themes having both Light
and Dark Appearance.
Change wording from Mode to Appearance, since Apple has NSAppearance.

rmottola commented:
Why was this closed without explanation?

rmottola committed:
Merge branch 'GSTheme_darkmode_branch' of https://github.com/gnustep/libs-gui into GSTheme_darkmode_branch

gcasa commented:
> Why was this closed without explanation?

Because I wasn't sure if this was the right approach.   Are you sure this is the right way to go?

Some of this functionality needs to reside in the backend as well.   The issue is that the theme is affecting how printing is processed.   It is a mystery to me how macOS handles this because some apps print exactly what is in the view and some don't.  So pinning down exactly what is done at the basic level is difficult without us writing our own word processing application which is no small feat.
